### PR TITLE
[INFRA-575] create workflows for merges & releases

### PR DESCRIPTION
**Pull request by eric-gro**
Create separate workflows for:
- pull requests
- merges into development branch
- release tags

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -77,42 +78,105 @@ jobs:
...
+          <<: *build-docs
+      - run:
+          <<: *test-docs
+      - run:
+          <<: *publish-docs
+      - run:
+          <<: *build-package
       # Publish new releases to PyPI.
       # We only check whether this is a tagged build and if so, publish to pypi
       # Check that $CIRCLE_TAG is not a null string
```
Comment (jsbrucker) [inline .circleci/config.yml]: Does this comment need to be updated?

File: .circleci/config.yml [hunk 2]
```diff
@@ -77,42 +78,105 @@ jobs:
...
+      - run:
+          <<: *test-docs
+      - run:
+          <<: *publish-docs
+      - run:
+          <<: *build-package
       # Publish new development updates to TestPyPI whenever changes are merged
       # to development.
       # We only want to publish a new package when PRs are actually accepted and
       # merged, which is why we check $CIRCLE_PULL_REQUEST.
```
Comment (jsbrucker) [inline .circleci/config.yml]: Ditto

File: .circleci/config.yml [hunk 3]
```diff
@@ -77,42 +78,105 @@ jobs:
...
+          filters:
+            branches:
+              only: "/^development$/"
+
+  release:
+    jobs:
+      - release-build:
+          filters:
+            branches:
+              ignore: "/.*/"
```
Comment (jsbrucker) [inline .circleci/config.yml:179]: Does this signify ignore changes to every branch?
Combined with the tags line does it mean only run when a tag is added?

File: .circleci/config.yml (hunk 1 above)
Comment (eric-gro) [reply .circleci/config.yml]: Yes.. needs an update. Need CircleCI to build for a PR first though. Which is has.

File: .circleci/config.yml (hunk 3 above)
Comment (eric-gro) [reply .circleci/config.yml:179]: Yes, combined it means only run with every new tag.

**Comment by eric-gro** [comment]
@muzigao do we want to push docs every time we merge to development or every time we create a new release tag?

**Review by gro-raywan — APPROVED** [review]
LGTM

**Review by John-Desmond — APPROVED** [review]
LGTM

**Review by muzigao-gro — APPROVED** [review]

**Comment by muzigao-gro** [comment]
> @muzigao do we want to push docs every time we merge to development or every time we create a new release tag?

Right now the doc site is by default showing development version. There's an selection you can toggle between release tags I think.  

We might want to revisit the behavior of dev doc (by default showing latest Tag), but can we keep the current behavior until we have the bandwidth?

**Review by fantastic-mr-fox — APPROVED** [review]
LGTM

**Review by Mbui-John — APPROVED** [review]

**Pull request merged**